homeassistant/components/deako/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""The deako integration."""
+from __future__ import annotations
+
+import logging
+
+from pydeako.deako import Deako, FindDevicesTimeout
+from pydeako.discover import DeakoDiscoverer
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up deako."""
+    _zc = await zeroconf.async_get_instance(hass)
+    discoverer = DeakoDiscoverer(_zc)
+
+    connection = Deako(discoverer.get_address)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = connection
```

Comment:
Can we move this to under `ConfigEntryNotReady`? This way we only populate the `hass.data` when everything succeeds.